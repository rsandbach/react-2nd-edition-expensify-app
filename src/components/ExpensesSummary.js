import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectTotalExpenses from '../selectors/expenses-total'

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => (
    <div>
        <h1>Viewing {expenseCount} {expenseCount > 1 ? 'expenses' : 'expense'} totalling {numeral(expenseTotal / 100.0).format('$0,0.00')}</h1>
    </div>
)

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectTotalExpenses(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)