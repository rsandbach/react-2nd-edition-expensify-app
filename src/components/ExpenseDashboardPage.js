import React from 'react'
import ExpenseList from '../components/ExpenseList'
import ExpenseListFilters from '../components/ExpenseListFilters'
import ExpensesSummary from '../components/ExpensesSummary'

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpensesSummary />
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage