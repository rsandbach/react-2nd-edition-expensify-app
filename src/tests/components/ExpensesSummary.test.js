import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpenseSummary correctly for one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={100} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary correctly for two expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={1100} />)
    expect(wrapper).toMatchSnapshot()
})