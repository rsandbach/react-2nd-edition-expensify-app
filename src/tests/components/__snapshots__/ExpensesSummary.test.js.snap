// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly for one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </p>
</div>
`;

exports[`should render ExpenseSummary correctly for two expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totalling 
    $11.00
  </p>
</div>
`;
